src/main/groovy/core/exceptions/EntityNotFoundException.groovy

```diff
@@ -0,0 +1,7 @@
+package core.exceptions
+
+class EntityNotFoundException extends DomainException {
```

Comment:
mesma coisa aqui, se cada validação for necessário uma exception específica, vão ter inúmeras a ponto de ficar dificil escolher qual usar em algum momento.